torch/csrc/jit/passes/dead_code_elimination.cpp

```diff
@@ -119,7 +119,7 @@ class DeadCodeEliminator {
       // TODO(suo): We shouldn't really have to check whether a node has uses,
       // since the mark algorithm should do that. But currently, the marking
       // doesn't reach loop counters in certain cases (see TestScript.test_pass)
```

Comment:
@suo what is the problem with TestScript.test_pass? It looks like DCE should clean up everything (including the loop) except for the constant: https://github.com/pytorch/pytorch/blob/ba70cf22fa867ada25f0dace95868999e53ef91f/test/test_jit.py#L8855-L8866

Reply:
It's because we currently don't have a good way of handling loop carried dependencies in DCE (since they don't obey normal use-def rules).